src/eddymotion/model/base.py

```diff
@@ -521,6 +522,72 @@ class DKIModel(BaseDWIModel):
     _model_class = "dipy.reconst.dki.DiffusionKurtosisModel"
 
 
+class GaussianProcessModel(BaseModel):
+    """A Gaussian process model for DWI data based on [Andersson15]_."""
+
+    __slots__ = (
+        "_dwi",
+        "_kernel",
+        "_gpr",
+    )
+
+    def __init__(self, dwi, kernel, **kwargs):
```

Comment:
I think we can initalize with gtab and avoid dragging the dwi data within the object.
```suggestion
    def __init__(self, gtab, kernel, **kwargs):
```